### PR TITLE
input: fix cursor's jump to end of input when user edit text at the middle of input

### PR DESCRIPTION
**Pull request by persidskiy**
Solve #388 

/cc @dfilatov @narqo

### CODE REVIEW

File: common.blocks/input/input.js [hunk 1]
```diff
@@ -55,11 +55,17 @@ provide(BEMDOM.decl(this.name, {
...
     setVal : function(val, data) {
+        var control = this.elem('control');
+
         val = String(val);
 
         if(this._val !== val) {
             this._val = val;
-            this.elem('control').val(val);
+
+            if(control.val() !== val) {
```
Comment (dfilatov) [inline common.blocks/input/input.js]: We use `&&` for short conditions.

``` js
control.val() !== val && control.val(val);
```

Comment (persidskiy) [inline common.blocks/input/input.js]: Ok

File: common.blocks/input/input.js [hunk 2]
```diff
@@ -55,11 +55,17 @@ provide(BEMDOM.decl(this.name, {
      * @returns {this}
      */
     setVal : function(val, data) {
+        var control = this.elem('control');
```
Comment (dfilatov) [inline common.blocks/input/input.js]: Seems that variable `control` is used only inside the next `if`. It should be declared nearer to place of using.

Comment (persidskiy) [inline common.blocks/input/input.js]: I had thought same, but decided to push all `var` declarations to top of the function.  :)

**Comment by persidskiy** [comment]
:up:

File: common.blocks/input/input.js [hunk 3]
```diff
@@ -58,8 +58,12 @@ provide(BEMDOM.decl(this.name, {
         val = String(val);
 
         if(this._val !== val) {
+            var control = this.elem('control');
```
Comment (dfilatov) [inline common.blocks/input/input.js]: And the final touch -- move this declaration to place before its first using.

**Comment by persidskiy** [comment]
done

**Pull request merged**